package.json

```diff
@@ -40,7 +40,6 @@
     "@types/react-test-renderer": "16.8.1",
     "@types/react-tooltip": "^3.3.5",
     "@types/react-transition-group": "2.9.1",
-    "@types/styled-components": "^4.1.16",
```

Comment:
What's wrong with `@types/styled-components`?

Reply:
Ah~ TS 3.9 does not allow | operator in an extend clause like this
```
export type StyledComponentInnerComponent<
    C extends React.ComponentType<any>
> = C extends
    | StyledComponent<infer I, any, any, any>
    | StyledComponent<infer I, any, any>
    ? I
    : C;
```

So I just change it to
```
export type StyledComponentInnerComponent<
    C extends React.ComponentType<any>
> = C extends StyledComponent<infer I, any, any, any> ? I :
    C extends StyledComponent<infer I, any, any> ? I :
    C;
```

^ is fixed in @types/styled-components@5.0.1 I think, but I guess we don't wanna use that